handlers/meshery_application_handler.go

```diff
@@ -122,17 +125,43 @@ func (h *Handler) handleApplicationPOST(
 		mesheryApplication := parsedBody.ApplicationData
 
 		bytApplication := []byte(mesheryApplication.ApplicationFile)
-		// check whether the uploaded file is a docker compose file
-		err := kompose.IsManifestADockerCompose(bytApplication, "")
-		if err == nil {
-			res, err := kompose.Convert(bytApplication) // convert the docker compose file into kubernetes manifest
+		mesheryApplication.SourceContent = bytApplication
+		var composeerr = fmt.Errorf("not docker compose")
+		if sourcetype == "" {
```

Comment:
Do we need to support this?
This still makes the file type inference ambiguous. Should we not make it a required thing?

Reply:
The spec states that it should be optional. Let's carry this discussion on the spec.

Reply:
Spec update: Given our outstanding challenges, this parameter is to be required.

Reply:
@leecalcote  thank you for clarification. I have made the query param required now.